components/hierarchical-view/hierarchical-view-mixin.js

```diff
@@ -0,0 +1,380 @@
+import { findComposedAncestor, isComposedAncestor } from '../../helpers/dom.js';
+import { getNextFocusable, getPreviousFocusable } from '../../helpers/focus.js';
+import ResizeObserver from 'resize-observer-polyfill/dist/ResizeObserver.es.js';
+
+export const HierarchicalViewMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			shown: { type: Boolean, reflect: true },
+			childView: { type: Boolean, reflect: true, attribute: 'child-view' },
+			isAttached: { type: Boolean, reflect: true, attribute: 'is-attached' },
+			isHierarchicalView: { type: Boolean, reflect: true, attribute: 'is-hierarchical-view' }
+		};
+	}
+
+	constructor() {
+		super();
+
+		this.isHierarchicalView = true;
+		this.__nativeFocus = null;
+		this.__focusPrevious = false;
+		this.__resizeObserver = null;
+	}
+
+	firstUpdated() {
+		super.firstUpdated();
+		this.addEventListener('keyup', this.__onKeyUp);
+		this.addEventListener('d2l-hierarchical-view-hide-start', this.__onHideStart);
+		this.addEventListener('d2l-hierarchical-view-show-start', this.__onShowStart);
+		this.addEventListener('d2l-hierarchical-view-resize', this.__onViewResize);
+		this.__nativeFocus = document.createElement('div').focus;
+		this.__onWindowResize = this.__onWindowResize.bind(this);
+
+		const parentView = findComposedAncestor(
+			this.parentNode,
+			(node) => { return node.isHierarchicalView; }
+		);
+
+		if (parentView) {
+			this.childView = true;
```

Comment:
I think the original code did this initialization inside of `attached`. In theory, a root hierarchical-view could be detached from the DOM, and append to another hierarchical-view. However, by initializing `this.childView` in `firstUpdated`, it would not be reinitialized as we would expect in such a scenario.  It's perhaps not a scenario we expect, but I think it is one that would break since `firstUpdated` is only invoked once, regardless of how many times the element is connected/disconnected.